clientapi/routing/admin_whois.go

```diff
@@ -47,8 +47,7 @@ func GetAdminWhois(
 	req *http.Request, userAPI api.UserInternalAPI, device *api.Device,
 	userID string,
 ) util.JSONResponse {
-	if userID != device.UserID {
-		// TODO: Still allow if user is admin
+	if device.AccountType != api.AccountTypeAdmin && userID != device.UserID {
```

Comment:
I think it would be clearer inverted. Access controls checks are [easy to mess-up](https://owasp.org/Top10/A01_2021-Broken_Access_Control/) so splitting this out to something like the following would be nicer I think:
```go
allowed := device.AccountType == api.AccountTypeAdmin || userID == device.UserID
if !allowed {
    return ....
}
```

Reply:
Definitely makes it easier to read.